packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -358,6 +385,26 @@ export class FileNavigatorContribution extends AbstractViewContribution<FileNavi
                 });
             }
         });
+        registry.registerCommand(OpenEditorsCommands.CLOSE_ALL_TABS_FROM_TOOLBAR, {
+            execute: widget => this.withOpenEditorsWidget(widget, async () => {
+                this.shell.widgets.forEach(w => {
```

Comment:
I know we've talked about this a bit, but I think perhaps the `.all` field on the `EditorManager` would give you these? It's supposed to track all the widgets created by the `EditorManager` that are still open, which should include those that are open inside an `EditorPreviewWidget`. But you may have already told me why that doesn't work...

Reply:
I had originally used that but ended up favoring the shell approach since It's possible in the future we might want to include widgets other than EditorWidgets in the tree. Ultimately I just decided to remove the dependency on the Editor package and using `NavigatableWidget.is` instead.